lib/addons/adot/index.ts

```diff
@@ -17,9 +17,9 @@ const versionMap: Map<KubernetesVersion, string> = new Map([
 /**
  * Configuration options for the Adot add-on.
  */
-export type AdotCollectorAddOnProps = Omit<CoreAddOnProps, "saName" | "addOnName" > & {
-    namespace: string; 
-  };
+export type AdotCollectorAddOnProps = Partial<Omit<CoreAddOnProps, "addOnName" | "controlPlaneAddOn" | "saName">> & {
```

Comment:
Why is this a `controlPlaneAddOn` and also you are missing `version` in omit and also in `AdotCollectorAddOnProps` which is the fundamental fix for the problem.

Reply:
Yes because you have not added it from Omit as i requested. Also you need to add that as field to CollectorPros. Check this code from Patrick

https://github.com/aws-quickstart/cdk-eks-blueprints/pull/975#issuecomment-2035181472

Reply:
```
import { KubernetesManifest } from "aws-cdk-lib/aws-eks";
import { Construct, IConstruct } from 'constructs';
import { ClusterInfo } from "../../spi";
import { createNamespace, dependable, loadYaml, readYamlDocument, supportsALL } from "../../utils";
import { CertManagerAddOn } from "../cert-manager";
import { CoreAddOn, CoreAddOnProps } from "../core-addon";
import { getAdotCollectorPolicyDocument } from "./iam-policy";
import { KubernetesVersion } from "aws-cdk-lib/aws-eks";

const versionMap: Map<KubernetesVersion, string> = new Map([
    [KubernetesVersion.V1_29, "v0.92.1-eksbuild.1"],
    [KubernetesVersion.V1_28, "v0.92.1-eksbuild.1"],
    [KubernetesVersion.V1_27, "v0.92.1-eksbuild.1"],
    [KubernetesVersion.V1_26, "v0.92.1-eksbuild.1"],
]);

/**
 * Configuration options for the Adot add-on.
 */
export type AdotCollectorAddOnProps = Omit<CoreAddOnProps, "saName" | "addOnName" | "version"> & {
    namespace: string;
    version?: string; 
  };

const defaultProps = {
    addOnName: 'adot',
    version: 'auto',
    versionMap: versionMap,
    saName: 'adot-collector',
    policyDocumentProvider: getAdotCollectorPolicyDocument,
    namespace: 'default',
    configurationValues: {}
};

 /**
  * Implementation of Adot Collector EKS add-on.
  */
@supportsALL
export class AdotCollectorAddOn extends CoreAddOn {
    private _version: string;

    constructor(props?: AdotCollectorAddOnProps) {
        super({ 
            ...defaultProps,
            namespace: props?.namespace ?? defaultProps.namespace,
            version: props?.version ?? defaultProps.version, 
            ...props
        });

        this._version = props?.version ?? 'auto';

    }

    public get version() {
        return this._version;
    }

    @dependable(CertManagerAddOn.name)
    deploy(clusterInfo: ClusterInfo): Promise<Construct>  {
        
        const addOnPromise = super.deploy(clusterInfo);
        return addOnPromise;
    }

    /**
     * Overriding base class method to create namespace and register permissions.
     * @param clusterInfo 
     * @param name 
     * @returns 
     */
    createNamespace(clusterInfo: ClusterInfo, namespaceName: string): IConstruct | undefined {
        // Create namespace if not default
        const cluster = clusterInfo.cluster;
        const ns = createNamespace(namespaceName, cluster, true, true);

        // Applying ADOT Permission manifest
        const otelPermissionsDoc = readYamlDocument(__dirname + '/otel-permissions.yaml');
        const otelPermissionsManifest = otelPermissionsDoc.split("---").map(e => loadYaml(e));
        const otelPermissionsStatement = new KubernetesManifest(cluster.stack, "adot-addon-otelPermissions", {
            cluster,
            manifest: otelPermissionsManifest,
            overwrite: true,
            
        });
        
        otelPermissionsStatement.node.addDependency(ns); 
        return otelPermissionsStatement;
    }
}
```

Reply:
Basically if you take a look in to [this](https://github.com/aws-quickstart/cdk-eks-blueprints/blob/01a78dcc2f5b54a41fe708fc4eaf862d07ad4a56/lib/addons/core-addon/index.ts#L20) `CoreAddOnProps`, the `version` field is mandatory. So in order for us to override the mandatory field to optional (so `auto` kicks in) we might have make this change so that some fields are omitted from being mandatory. So no inheritance to some properties of  `CoreAddOnProps`.

Reply:
I was talking more about consistency. Approach seems to be good. Let me run e2e if all works out well. Thankyou so much for detailed test cases @Falco152